[46기 김지연] ADD: Nav 구현

yeonn-k opened the pull request:
## 1. 본 PR이 우리 팀의 웹 서비스 제품성에 어떠한 기여를 하였고, 사용자에게 어떠한 기대효과를 전달하는지 작성해주세요.
- 내 PR이 제품 내 어떠한 기능적인 배경/전후맥락 가운데 개발되었나요?
- 고객 층의 연령대가 다양하므로, 심플하게 구성하되 브랜딩 컬러를 포인트 컬러로 사용하여 행복하고 따뜻한 느낌을 전달합니다.

- 내 PR이 Merge 됨으로써 유저에게 전달되는 편익/기대효과는 무엇일까요?
- 카테고리를 편하게 이동할 수 있습니다.

<br />

## 2. 이 브랜치에서 어떤 내용을 개발했는지 큰 제목과 그리고 상세 내역을 적어주세요.
- Nav bar 생성
- Nav 버튼 클릭시 모달창 구현 
   : ham 메뉴 css/ keyframe 적용, icon hover 시 컬러 변경

<br />

## 3. 개발한 화면을 캡쳐해서 첨부 해 주세요. (drag & drop 또는 첨부파일 추가)
<br />
![스크린샷 2023-05-31 오후 3 10 47](https://github.com/wecode-bootcamp-korea/46-1st-WePet-frontend/assets/130032082/783a0fef-2547-4d04-8ba2-da0a3e40a4f9)


<br />

## 4. 이 브랜치에서 개발하면서 느꼇던 개발 성장포인트를 적어주세요.

- ham 메뉴 keyframe 적용
- modal 창 투명도 있는 배경색 주기
- fontawesome icon = text
- position 체크!

Commit:
ADD: Nav 구현

Commit:
.

Commit:
Fix: 모달창 햄버거 메뉴 state 관리 수정

Commit:
ADD: modal X버튼 기능 구현

sstaar91 commented on src/components/Nav/Modal.js:
```diff
@@ -0,0 +1,68 @@
+          </Link>
+          <Link to="/">
+            <li>제 방 구경하실 분</li>
+          </Link>
```
해당 부분이 반복되는 UI네요!
상수데이터로 만들어서 구현할 수 있겠습니다!

sstaar91 commented on src/components/Nav/Modal.js:
```diff
@@ -0,0 +1,68 @@
+          </Link>
+          <Link to="/">
+            <li>용품</li>
+          </Link>
```
여기도 마찬가지로 상수데이터로 구현할 수 있겠습니다!

sstaar91 commented on src/components/Nav/Modal.js:
```diff
@@ -0,0 +1,68 @@
+            <div className="inquireIcon">
+              <FontAwesomeIcon
+                icon={faHeadset}
+                style={{ color: '#ffffff' }}
```
인라인 스타일은 최대한 지양해 주셔야 합니다!
해당 아이콘에 스타일을 적용하려고 한다면 className을 적용하시고
color 스타일로 적용해주세요~

Commit:
FIX: searchModal 기능구현

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,88 @@
-import React from 'react'
+import React, { useState } from 'react'
+import Modal from './Modal'
+import SearchModal from './SearchModal'
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons'
+import { faCartShopping, faHeadset } from '@fortawesome/free-solid-svg-icons'
+import { Link } from 'react-router-dom'
```
import 순서 수정해주세요! 일반적인 convention을 따라 순서만 잘 지켜주셔도 가독성이 좋아집니다. 
아래 순서 참고해주세요.

React → Library(Package) → Component → 변수 / 이미지 → css 파일(scss 파일)

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,88 @@
 const Nav = () => {
-  return <div>hello</div>
+  const [isHover, setIsHover] = useState(false)
+  const [modal, setModal] = useState(false)
```
위에 의문형으로 state 선언해 주신 것 처럼 초기값이 boolean일 경우 state의 이름을
의문형으로 적용해주세요!

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,88 @@
+          </div>
+        )}
+      </div>
+      {searchModal ? (
+        <SearchModal
+          searchModal={searchModal}
+          setSearchModal={setSearchModal}
+        />
+      ) : (
+        ''
+      )}
```
```suggestion
      {searchModal && (
        <SearchModal
          searchModal={searchModal}
          setSearchModal={setSearchModal}
        />}
```
조건부 렌더링 할 때 다른 false일 때 UI가 필요하지 않다면 && 연산자를 활용해주세요!

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,88 @@
+      ) : (
+        ''
+      )}
+      {modal === true ? <Modal /> : ''}
```
modal 값 자체가 boolean 값이기에 true와 비교하지 않아도 되고,
여기도 && 연산자를 활용해주세요!

sstaar91 commented on src/components/Nav/Nav.scss:
```diff
@@ -0,0 +1,302 @@
+@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+KR:wght@300&display=swap');
+@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@500&display=swap');
+
+a {
+  text-decoration: none;
+}
```
이렇게 태그에 직접적으로 적용하는 것은 모든 페이지의 a 태그에 해당 스타일이 적용될 수 있습니다.
모든 a 태그에 적용되어야 한다면 common 혹은 reset에서 적용해주세요!

sstaar91 replied:
이 부분은 footer가 merge 되면 지우셔도 되겠네요!

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,88 @@
+
+  return (
+    <>
+      <div className="Nav">
```
className은 소문자로 시작해야 합니다!

sstaar91 commented on src/components/Nav/Nav.scss:
```diff
@@ -0,0 +1,302 @@
+  text-decoration: none;
+}
+
+.Nav {
```
여기도 변경되어야 하겠네요!

sstaar91 commented on src/components/Nav/Nav.scss:
```diff
@@ -0,0 +1,302 @@
+  height: 50px;
+  line-height: 50px;
+  padding: 7px;
+  // background-color: #ffbb0f;
```
주석 제거해주세요~

sstaar91 commented on src/components/Nav/Nav.scss:
```diff
@@ -0,0 +1,302 @@
+  .category {
+    width: 100%;
+    // height: 50px;
+    span {
+      color: #000;
+      font-size: 13px;
+      margin-right: 16px;
```
빈번히 사용되는 태그로 스타일을 주는 건 depth에 무관하게 적용되어 의도치 않은 스타일링이 될 수 있습니다. 
tag selector는 해당 요소 뿐만 아니라 동일한 태그로 선언된 다른 요소들에게도 영향을 
미칠 여지가 너무 많기 때문에 추후 유지보수에 악영향을 미칠 수 있습니다.
스타일링을 줄 때는 요소마다 className을 부여하셔서 스타일링을 주시는게 좋습니다.

다른 부분도 동일하게 적용한 부분이 있다면 className을 적용해주세요

sstaar91 commented on src/components/Nav/Nav.scss:
```diff
@@ -0,0 +1,302 @@
+    color: #000;
+  }
+
+  .modalBottom {
+    position: fixed;
+    width: 260px;
+    height: 80px;
+    display: flex;
+    align-items: center;
+    bottom: 0;
+    right: 0;
+  }
```
컴포넌트 분리하신것처럼 스타일도 분리해서 관리해주세요!

sstaar91 reviewed (commented):
지연님 수고하셨습니다!
수정 사항 반영해주세요~

sstaar91 reviewed (commented):
지연님 수고하셨습니다!
수정 사항 반영해주세요~

Commit:
FIX: 리뷰사항 반영
1. 컴포넌트 별 scss 분리
2. className 수정
3. 반복되는 UI 상수데이터로 구현
4. 인라인 스타일 제거
5. import 순서 convention 적용
6. boolean state 이름 의문형으로 변경

Commit:
FIX: hover시 cursor변경 일부 추가

Commit:
FIX: 조건부 렌더링 삼항연산자 => && 연산자로 변경

Commit:
FIX: 반복데이터=> 상수데이터로 관리 / map

sstaar91 commented on src/components/Nav/Modal.js:
```diff
@@ -0,0 +1,56 @@
+import { Link } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faHeadset } from '@fortawesome/free-solid-svg-icons'
+import MODAL_DATA1 from './modalData'
+import MODAL_DATA2 from './modalData2'
```
여기 data도 Footer에 있는 내용과 동일하게 구현해주세요!

sstaar91 commented on src/components/Nav/Modal.js:
```diff
@@ -0,0 +1,56 @@
+        </div>
+        <span className="ulTitle">테마</span>
+        <ul className="ulBox">
+          {MODAL_DATA1.map(data => {
```
변경한 이름에 따라서 다른 데이터 이름이 들어가야겠네요!

sstaar91 commented on src/components/Nav/Modal.scss:
```diff
@@ -0,0 +1,116 @@
+    font-size: 13px;
+    color: #000;
+
+    // :hover {
+    //   color: #ffbb0f;
+    // }
```
주석은 제거해주세요!

sstaar91 commented on src/components/Nav/Modal.scss:
```diff
@@ -0,0 +1,116 @@
+  font-weight: 900;
+  animation: modalShowRight 0.3s;
+
+  @keyframes modalShowRight {
+    from {
+      opacity: 0;
+      margin-right: -50px;
+    }
+    to {
+      opacity: 1;
+      margin-right: 0;
+    }
+  }
```
keyframe 같은 경우는 따로 애니메이션 스타일 파일을 만들어서 관리해 주는 것이 좋습니다.

sstaar91 commented on src/components/Nav/Modal.scss:
```diff
@@ -0,0 +1,116 @@
+  }
+  p {
+    padding-bottom: 8px;
+  }
```
빈번히 사용되는 태그로 스타일을 주는 건 depth에 무관하게 적용되어 의도치 않은 스타일링이 될 수 있습니다. 
tag selector는 해당 요소 뿐만 아니라 동일한 태그로 선언된 다른 요소들에게도 영향을 
미칠 여지가 너무 많기 때문에 추후 유지보수에 악영향을 미칠 수 있습니다.
스타일링을 줄 때는 요소마다 className을 부여하셔서 스타일링을 주시는게 좋습니다.

다른 부분도 동일하게 적용한 부분이 있다면 className을 적용해주세요

sstaar91 commented on src/components/Nav/Modal.scss:
```diff
@@ -0,0 +1,116 @@
+    margin-bottom: 10px;
+    margin: 6px 0px;
+    font-size: 13px;
+    color: #000;
```
이 부분은 nesting 해서 사용할 수 있겠네요!

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,91 @@
 import React from 'react'
+import { useState } from 'react'
```
```suggestion
import React, {useState} from 'react'
```
같은 위치에서 불러오는 거라면 이렇게 쉼표를 사용해서 불러올 수 있습니다!

sstaar91 commented on src/components/Nav/SearchModal.js:
```diff
@@ -0,0 +1,34 @@
+            setSearchModal(!searchModal)
+          }}
+        >
+          <div className="line hamTopLine"></div>
+          <div className="line hamBtmLine"></div>
```
self closing 해주세요!

sstaar91 commented on src/components/Nav/SearchModal.scss:
```diff
@@ -0,0 +1,85 @@
+        opacity: 1;
+        margin-top: 0;
+      }
+    }
```
위에서 말씀드린 것 처럼 애니메이션 스타일 파일을 만들어서 관리해주세요!

sstaar91 commented on src/components/Nav/SearchModal.scss:
```diff
@@ -0,0 +1,85 @@
+
+      input {
+        width: 475px;
+        font-size: 19px;
```
빈번히 사용되는 태그로 스타일을 주는 건 depth에 무관하게 적용되어 의도치 않은 스타일링이 될 수 있습니다. 
tag selector는 해당 요소 뿐만 아니라 동일한 태그로 선언된 다른 요소들에게도 영향을 
미칠 여지가 너무 많기 때문에 추후 유지보수에 악영향을 미칠 수 있습니다.
스타일링을 줄 때는 요소마다 className을 부여하셔서 스타일링을 주시는게 좋습니다.

다른 부분도 동일하게 적용한 부분이 있다면 className을 적용해주세요

sstaar91 reviewed (commented):
지연님 수고하셨습니다!
수정 사항 반영해주세요~

Commit:
FIX: 리뷰사항 반영
1. data 이름 수정
2. 주석제거
3. 애니메이션 스타일 => animation.scss 파일 별도 작성
4. tag css 적용  => className css 적용
5. div 태그 self closing

Commit:
FIX: scss nesting 적용

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,92 @@
+          <FontAwesomeIcon
+            icon={faMagnifyingGlass}
+            onClick={() => {
+              setIsSearchModal(!isSearchModal)
```
```suggestion
              setIsSearchModal(prev => !prev)
```
함수형 업데이트를 적용해주시는게 더 좋습니다~!

sstaar91 commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,92 @@
+          <div
+            className="ham"
+            onClick={() => {
+              setIsModal(!isModal)
```
```suggestion
              setIsModal(prev => !prev)
```
여기도 이렇게 적용해주세요!

sstaar91 reviewed (commented):
지연님 하나만 수정 반영해주세요~

Commit:
FIX: 리뷰사항 반영( 함수형 업데이트 적용), Nav Menu CSS 수정

Commit:
FIX: logo 및 font-size 수정

Commit:
ADD: text-decoration:none 추가

Commit:
FIX:Nav 수정

Commit:
FIX: conflict 해결

sstaar91 reviewed (approved):
LGTM 🚀🚀🚀🚀
